### PR TITLE
added redirect functionality to Routes

### PR DESCRIPTION
**Pull request by adamlangsner**
@samandmoore @dschaub 

added redirect functionality to route filters

### CODE REVIEW

**Comment by samandmoore** [comment]
whoa! that's great. thanks for diving into this.

some feedback:
- i think we want to be able to support redirects from inside of routes as well as from filters, and it would be nice for both of those mechanisms to share as much code as possible.
- following up on the first point, i think that means that we might want to have this method set that redirect value on `routeData` and then we check `routeData` instead of the filter return value.

those are my initial thoughts. will look at the code a bit more later.

**Comment by adamlangsner** [comment]
@samandmoore if we want to be able to redirect within `prepare`, `execute` and `error`. why don't we set the router as an ivar on the each route so that they can just do `this.router.navigate();`?

**Comment by adamlangsner** [comment]
@samandmoore @dschaub bump
i added redirection functionality to prepare, execute and error. they can now all do `return this.redirect('new_path');`.

I wanted to keep the pattern where we return the redirect, instead of setting state on routeData because it requires that you return in order to do a redirect, which prevents us from forgetting to return.

@samandmoore what are the benefits of setting state on routeData as compared to checking the return value of the filter, exectue, prepare and error methods

**Comment by samandmoore** [comment]
i think you're right. i have no arguments against this solution anymore.

File: backbone.blazer.js [hunk 1]
```diff
@@ -60,19 +65,19 @@ Backbone.Blazer.Router = Backbone.Router.extend({
         route.trigger('before:execute', routeData);
         router.trigger('before:execute', routeData);
 
-        this._runBeforeFilters(route, routeData).then(function() {
-            return $.when(route.prepare(routeData));
-        }).then(function() {
+        this._runBeforeFilters(router, route, routeData).then(function() {
+            return this._runHandler(route.prepare, router, route, routeData);
+        }.bind(this)).then(function() {
```
Comment (samandmoore) [inline backbone.blazer.js]: i've been trying to avoid using `.bind(this)` as much as possible in this code because i think it makes things less readable.

Comment (dschaub) [inline backbone.blazer.js]: yeah, we're using the `router` alias for `this` in other parts of these callbacks already too

**Comment by adamlangsner** [comment]
@dschaub @samandmoore 
removed the `bind(this)`s

**Comment by adamlangsner** [comment]
BUMP. updated the readme to include info on filters and redirects.

**Comment by dschaub** [comment]
nice job on the readme, I think this looks good :+1:

**Pull request merged**